### DISCUSSION TITLE
[WASM] Development cycle using Firefox

### DISCUSSION THREAD

**Discussion by DierkDroth** [unoplatform/uno#12377] [Q&A]
I know that I should use Chrome to develop an UNO WASM app - that's what I'm doing since years.

However, I just came across an issue - presumably in my Javascript code around the iframe-based WebView2 'simulation' which I have in place - which only shows in Firefox (likely some nasty differences in JavaScript event handling between Chrome and Firefox).

So, I was wondering whether there is a way to have a (relatively) quick development cycle to test my UNO WASM code changes with Firefox as there is with Chrome. Going through full rebuilds + uploads to a my server(s) is just too time consuming... 

Any hint would be appreciated. Thanks in advance

**Comment by jeromelaban** [comment]
What's the issue about debugging with firefox? Visual Studio is able to select Firefox in the debugger selector.

**Reply by DierkDroth** [reply]
Oh, I didn't know that. I'll look into... thanks

**Reply by DierkDroth** [reply]
@jeromelaban hmm ... I did [some research ](https://learn.microsoft.com/en-us/aspnet/core/blazor/debug?view=aspnetcore-7.0&tabs=visual-studio)and also consulted Bing chat ;-) ... appears only Chrome/Edge are supported. I could not find any pointer on how to debug my UNO WASM sources right from Visual Studio

What am I missing?

**Reply by jeromelaban** [reply]
The official debugger documentation is the correct source. If it mentions it's not supported, then it's not supported by Uno either. The runtime has references to Firefox debugging, but it may not have been released publicly.

**Comment by DierkDroth** [comment]
Thanks for confirmation @jeromelaban. 

So to wrap this up: no way to debug UNO WASM source using Firefox
